### PR TITLE
[libraries/pod]: improve `PodOption` type

### PR DESCRIPTION
**Pull request by febo**
### Problem

The `PodOption` type currently has 2 limitations:
1. It does not handle correctly the conversion of `Option::Some(T::default())` to `PodOption` since `T::default()` corresponds to the `None` value. As a result, it will convert a `Some` to a `None` value.
2. It always assume that `T:default()` is the `None` value, which might not be true in all cases and could be a limitation.

### Solution

To address (1), replace the `From` implementation for a `TryFrom`, where the conversion will fail when trying to convert a `Option::Some(T::default())` value to `PodOption`.

To address (2), add a new associated constant `NONE` to the `Nullable` trait instead of relying on the `default` value. Types implementing the trait can then specify the value that represents `None`.

### CODE REVIEW

File: libraries/pod/src/option.rs [hunk 1]
```diff
@@ -76,32 +93,33 @@ impl<T: Nullable> From<T> for PodOption<T> {
...
-            Some(value) => PodOption(value),
-            None => PodOption(T::default()),
+impl<T: Nullable> TryFrom<Option<T>> for PodOption<T> {
+    type Error = ProgramError;
+
+    fn try_from(value: Option<T>) -> Result<Self, Self::Error> {
+        match value {
+            Some(value) if value.is_some() => Ok(PodOption(value)),
+            None => Ok(PodOption(T::NONE)),
+            _ => Err(ProgramError::InvalidArgument),
```
Comment (joncinque) [inline libraries/pod/src/option.rs:103]: nit: this is totally fine, but how about making this exhaustive to be safe?
```suggestion
            Some(value) if value.is_none() => Err(ProgramError::InvalidArgument),
            Some(value) => Ok(PodOption(value)),
            None => Ok(PodOption(T::NONE)),
```

**Review by joncinque — COMMENTED** [review]
Looks good! Just a couple of style nits

File: libraries/pod/src/option.rs (hunk 1 above)
Comment (febo) [reply libraries/pod/src/option.rs]: Nice!

**Review by joncinque — APPROVED** [review]
Looks great

**Pull request merged**